splitio/models/grammar/matchers/semver.py

```diff
@@ -0,0 +1,158 @@
+"""Semver matcher classes."""
+import logging
+
+_LOGGER = logging.getLogger(__name__)
+
+class Semver(object):
+    """Semver class."""
+
+    _METADATA_DELIMITER = "+"
+    _PRE_RELEASE_DELIMITER = "-"
+    _VALUE_DELIMITER = "."
+
+    def __init__(self, version):
+        """
+        Class Initializer
+
+        :param version: raw version as read from splitChanges response.
+        :type version: str
+        """
+        self._major = 0
+        self._minor = 0
+        self._patch = 0
+        self._pre_release = []
+        self._is_stable = False
+        self.version = ""
+        self._metadata = ""
+        self._parse(version)
+
+    @classmethod
+    def build(cls, version):
+        try:
+            self = cls(version)
+        except RuntimeError as e:
+            _LOGGER.error("Failed to parse Semver data, incorrect data type:  %s", e)
+            return None
+
+        return self
+
+    def _parse(self, version):
+        """
+        Parse the string in self.version to update the other internal variables
+        """
+        without_metadata = self.remove_metadata_if_exists(version)
+
+        index = without_metadata.find(self._PRE_RELEASE_DELIMITER)
+        if index == -1:
+            self._is_stable = True
+        else:
+            pre_release_data = without_metadata[index+1:]
+            without_metadata = without_metadata[:index]
+            self._pre_release = pre_release_data.split(self._VALUE_DELIMITER)
```

Comment:
I think here you should throw an error if some item in the `_pre_release` array is an empty string

Reply:
will do